### PR TITLE
Allow to filter bookings by booked_for/booking_reason

### LINKED ISSUE #3959: Room Booking: being able to search by "booked for"
It's currently not possible to search for a booking using its owner's name (search field in filter bar).
This is needed especially for Burotel but is of general interest.

### PR DESCRIPTION
**Pull request by mic4ael**
closes #3959

### CODE REVIEW

File: indico/modules/rb/operations/bookings.py [hunk 1]
```diff
@@ -156,9 +156,10 @@ def get_room_candidates(candidates, conflicts):
             if not (any(candidate.overlaps(conflict) for conflict in conflicts))]
 
 
-def _bookings_query(filters):
+def _bookings_query(filters, noload_room=False):
     reservation_strategy = contains_eager('reservation')
-    reservation_strategy.noload('room')
+    if noload_room:
+        reservation_strategy.noload('room')
```
Comment (ThiefMaster) [inline indico/modules/rb/operations/bookings.py]: I'd rather use `raiseload` so it fails loudly if there's code accessing it anyway.

File: indico/modules/rb/operations/bookings.py [hunk 2]
```diff
@@ -193,14 +203,16 @@ def get_room_calendar(start_date, end_date, room_ids, include_inactive=False, **
...
     query = query.order_by(db.func.indico.natsort(Room.full_name))
     rooms = (Room.query
-             .filter(~Room.is_deleted, Room.id.in_(room_ids) if room_ids else True)
+             .filter(~Room.is_deleted, Room.id.in_(room_ids))
              .options(joinedload('location'))
              .order_by(db.func.indico.natsort(Room.full_name))
              .all())
 
+    bookings = query.all()
+    rooms.extend(set(b.reservation.room for b in bookings if b.reservation.room not in rooms))
```
Comment (ThiefMaster) [inline indico/modules/rb/operations/bookings.py]: you can use a set literal here

Comment (ThiefMaster) [inline indico/modules/rb/operations/bookings.py]: what's the reason for the `not in rooms` check here? Just make `rooms` a set :) Sorting is broken here anyway after adding stuff to the list, so you need to re-sort anyway (and you can remove the `order_by` then)

Comment (mic4ael) [reply indico/modules/rb/operations/bookings.py]: I think it was due to some assumption I made initially which turned out to be incorrect. I think it can be removed altogether

File: indico/modules/rb/operations/bookings.py [hunk 3]
```diff
@@ -191,16 +202,16 @@ def get_room_calendar(start_date, end_date, room_ids, include_inactive=False, **
     end_dt = datetime.combine(end_date, time(hour=23, minute=59))
     query = _bookings_query(dict(filters, start_dt=start_dt, end_dt=end_dt, room_ids=room_ids,
                                  include_inactive=include_inactive))
-    query = query.order_by(db.func.indico.natsort(Room.full_name))
+    bookings = query.order_by(db.func.indico.natsort(Room.full_name)).all()
     rooms = (Room.query
-             .filter(~Room.is_deleted, Room.id.in_(room_ids) if room_ids else True)
+             .filter(~Room.is_deleted, Room.id.in_(room_ids))
```
Comment (ThiefMaster) [inline indico/modules/rb/operations/bookings.py]: why did you remove it? if `room_ids` is None this will now fail (easy to reproduce by just sending a GET request to the calendar endpoint so no room ids are included)

File: indico/modules/rb/operations/bookings.py [hunk 4]
```diff
@@ -191,16 +202,16 @@ def get_room_calendar(start_date, end_date, room_ids, include_inactive=False, **
...
              .options(joinedload('location'))
-             .order_by(db.func.indico.natsort(Room.full_name))
              .all())
 
+    rooms = sorted(set(rooms) | {b.reservation.room for b in bookings}, key=lambda r: natural_sort_key(r.full_name))
```
Comment (ThiefMaster) [inline indico/modules/rb/operations/bookings.py]: this is causing query spam (search e.g. for "meeting")
